src/solvers/changeCobraSolver.m

```diff
@@ -1,365 +1,385 @@
+    case 'dqqMinos'
+        solverOK = checkSolverInstallationExecutable(solverName, 'run1DQQ', true, printLevel);
+    case 'opti'
+        optiSolvers = {'CLP', 'CSDP', 'DSDP', 'OOQP', 'SCIP'};
+        if ~isempty(which('checkSolver'))
+            availableSolvers = cellfun(@(x)checkSolver(lower(x)), optiSolvers);
+            fprintf('OPTI solvers installed currently: ');
+            fprintf(char(allLPsolvers(logical(availableSolvers))));
+            if ~any(logical(availableSolvers))
+                return;
+            end
+        end
+    case 'matlab'
+        solverOK = true;
+    otherwise
+        error(['Solver ' solverName ' not supported by COBRA Toolbox']);
+end
+
+% set solver related global variables
+if solverOK
+    varName = horzcat(['CBT_', solverType, '_SOLVER']);
+    eval([varName ' =  solverName;']);
+end
+
+end
+
+
+function solverOK = checkGurobiInstallation(solverName, fileName, printLevel)
+% Check Gurobi installation.
+%
+% Usage:
+%     solverOK = checkGurobiInstallation(solverName, fileName)
+%
+% Inputs:
+%     solverName: string with the name of the solver
+%     fileName:   string with the name of the file to look for
+%
+% Output:
+%     solverOK: true if filename exists, false otherwise.
+%
+
+    global GUROBI_PATH
+    solverOK = false;
+    if ~isempty(findstr(GUROBI_PATH, solverName))
+        if exist(fileName) == 2
+            solverOK = true;
+        elseif printLevel > 0
+            error('Solver %s is not installed!', solverName)
+        end
+    elseif printLevel > 0
+        error('Solver %s is not installed!', solverName)
+    end
+end
+
+
+function solverOK = checkSolverInstallationFile(solverName, fileName, printLevel)
+% Check solver installation by existence of a file in the Matlab path.
+%
+% Usage:
+%     solverOK = checkSolverInstallation(solverName, fileName)
+%
+% Inputs:
+%     solverName: string with the name of the solver
+%     fileName:   string with the name of the file to look for
+%
+% Output:
+%     solverOK: true if filename exists, false otherwise.
+%
+    solverOK = false;
+    if exist(fileName, 'file') == 2 || exist(fileName, 'file') == 3
+        solverOK = true;
+    elseif printLevel > 0
+        error('Solver %s is not installed!', solverName)
+    end
+end
+
+
+function solverOK = checkSolverInstallationExecutable(solverName, executableName, unix, printLevel)
+% Check Gurobi installation.
+%
+% Usage:
+%     solverOK = checkGurobiInstallation(solverName, fileName)
+%
+% Inputs:
+%     solverName: string with the name of the solver
+%     executableName:   string with the name of the executable to look for
+%     unix:
+%
+% Output:
+%     solverOK: true if executableName exists, false otherwise.
+%
+    solverOK = false;
+    if unix
+        if ~isunix && printLevel > 0
+            error('%s interface not implemented for non unix OS', solverName);
+        end
+    end
+    [status, cmdout] = system(['which ' executableName]);
```

Comment:
Yes, indeed, this should be within the `if` condition. The function `checkSolverInstallationExecutable` is only intended for unix-specific solver interfaces (at least for now). Good catch :+1: